pkg/kots/rego/kots-spec-opa-nonrendered.rego

```diff
@@ -553,6 +566,119 @@ lint[output] {
+    "message": message,
+    "path": config_file_path,
+    "field": field,
+    "docIndex": config_data.docIndex
+  }
+}
+
+# Check if repeatable ConfigOption has at least one template
+lint[output] {
+  config_option := config_options[_]
+  item := config_option.item
+  item.repeatable
+  item.template == null
+  field := concat(".", [config_option.field, "type"])
+  message := sprintf("Repeatable Config option \"%s\" is missing a template target", [string(item.name)])
+  output := {
+    "rule": "repeat-config-option-missing-template",
+    "type": "error",
+    "message": message,
+    "path": config_file_path,
+    "field": field,
+    "docIndex": config_data.docIndex
+  }
+}
+
+# Check if repeatable ConfigOption has a valuesByGroup field
+lint[output] {
+  config_option := config_options[_]
+  item := config_option.item
+  item.repeatable
+  not item.valuesByGroup
+  field := concat(".", [config_option.field, "type"])
+  message := sprintf("Repeatable Config option \"%s\" is missing the valuesByGroup field", [string(item.name)])
+  output := {
+    "rule": "repeat-config-option-missing-valuesByGroup",
+    "type": "error",
+    "message": message,
+    "path": config_file_path,
+    "field": field,
+    "docIndex": config_data.docIndex
+  }
+}
+
+# Check if repeatable ConfigOption has at least one group in valuesByGroup
+lint[output] {
+  config_option := config_options[_]
+  item := config_option.item
+  item.repeatable
+  item.valuesByGroup
+  item.valuesByGroup == null
+  field := concat(".", [config_option.field, "type"])
+  message := sprintf("Repeatable Config option \"%s\" is missing a valuesByGroup group", [string(item.name)])
+  output := {
+    "rule": "repeat-config-option-missing-valuesByGroup-group",
+    "type": "error",
+    "message": message,
+    "path": config_file_path,
+    "field": field,
+    "docIndex": config_data.docIndex
+  }
+}
+
+# Check if repeatable ConfigOption has at least one value in each valuesByGroup
+lint[output] {
+  config_option := config_options[_]
+  item := config_option.item
+  item.repeatable
+  item.valuesByGroup
+  valueGroup := item.valuesByGroup[_]
+  valueGroup == null
+  field := concat(".", [config_option.field, "type"])
+  message := sprintf("Repeatable Config option \"%s\" is missing a value for group", [string(item.name)])
+  output := {
+    "rule": "repeat-config-option-missing-valuesByGroup-value",
+    "type": "error",
+    "message": message,
+    "path": config_file_path,
+    "field": field,
+    "docIndex": config_data.docIndex
+  }
+}
+
+# Check if repeatable ConfigOption template ends in array
+lint[output] {
+  config_option := config_options[_]
+  item := config_option.item
+  item.repeatable
+  template := item.template[_]
+  template.yamlPath
+  not template_yamlPath_ends_with_array(template)
+  field := concat(".", [config_option.field, "type"])
+  message := sprintf("Repeatable Config option \"%s\" yamlPath does not end with an array", [string(item.name)])
```

Comment:
stupid question, why does the yamlPath have to end with an array?

Reply:
When we duplicate the yaml node, we need to append the duplicates into an array.  The yamlPath should point to the array node that needs to be duplicated, and the location of where duplicates are appended to.

It's possible for a future update to allow yamlPath to end at a map definition.  If we end up making that update we can remove this linting rule.